spec/order_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'order'
+
+describe Order do
+
+  describe "#initialize" do
+    it "should initialize with a menu array" do
+      menu_double = double :menu, new: Array 
+      order = Order.new

```

Comment:
I've had another look at this, and I'm not sure you've passed in the double to override the default argument. Wouldn't we need to say?:
order = Order.new(menu_double)

Reply:
Right... it does look strange, but the test fails any other way I tried. I wonder if it is to do with the dependency injection initializing a menu by default?